test/integration/api/external/index.js

```diff
@@ -0,0 +1,245 @@
+		it('typeof fn', function() {
+			assert.isFunction(axe.commons.aria.implicitRole);
+		});
+		it('el => str|null', function() {
+			var implicitRolesOrNull = getEntries(
+				axe.commons.aria.lookupTable.role
+			).reduce(
+				function(roles, entry) {
+					var role = entry[0];
+					var val = entry[1];
+					if (val.implicit) {
+						roles.push(role);
+					}
+					return roles;
+				},
+				[null]
+			);
+			var role = axe.commons.aria.implicitRole(randomNodeInTree());
+			assert.oneOf(role, implicitRolesOrNull);
+		});
+	});
+
+	describe('axe.commons.aria.lookupTable.role', function() {
+		it('typeof obj', function() {
+			assert.isObject(axe.commons.aria.lookupTable.role);
+		});
+		it('str <=> str (role.type)', function() {
+			var keys = getKeys(axe.commons.aria.lookupTable.role);
+			var types = getValues(axe.commons.aria.lookupTable.role).map(function(
+				role
+			) {
+				return role.type;
+			});
+			keys.forEach(assert.isString);
+			types.forEach(assert.isString);
+		});
+	});
+
+	describe('axe.utils.getFlattenedTree', function() {
+		it('typeof fn', function() {
+			assert.isFunction(axe.utils.getFlattenedTree);
+		});
+		it('el => arr', function() {
+			assert.isArray(axe.utils.getFlattenedTree(document.body));
+		});
+		it('el => [{root}]', function() {
+			assert.lengthOf(
+				axe.utils.getFlattenedTree(randomNodeInTree(isElement)),
+				1
+			);
+		});
+	});
+
+	describe('axe.utils.getNodeFromTree', function() {
+		it('typeof fn', function() {
+			assert.isFunction(axe.utils.getNodeFromTree);
+		});
+		it('node => vnode', function() {
+			axe._tree = axe.utils.getFlattenedTree(document.body);
+			assert.oneOf(
+				axe.utils.getNodeFromTree(randomNodeInTree()),
+				flat(axe._tree[0])
+			);
+		});
+		it('!node => ()', function() {
+			assert.isUndefined(axe.utils.getNodeFromTree(randomElement()));
+		});
+	});
+
+	describe('axe.commons.dom.isOpaque', function() {
+		it('typeof fn', function() {
+			assert.isFunction(axe.commons.dom.isOpaque);
+		});
+		it('el inTree => bool', function() {
+			assert.isBoolean(axe.commons.dom.isOpaque(randomNodeInTree(isElement)));
+		});
+		it('el => bool', function() {
+			assert.isBoolean(axe.commons.dom.isOpaque(randomElement()));
+		});
+	});
+
+	describe('axe.commons.text.accessibleTextVirtual', function() {
+		it('typeof fn', function() {
+			assert.isFunction(axe.commons.text.accessibleTextVirtual);
+		});
+		it('el vnode => str', function() {
+			axe._tree = axe.utils.getFlattenedTree(document.body);
+			var vnode = axe.utils.getNodeFromTree(randomNodeInTree(isElement));
+			assert.isString(axe.commons.text.accessibleTextVirtual(vnode));
+		});
+	});
+});
+
+var elements = [
+	document.createElement('div'),
+	document.createElement('button'),
+	document.createElement('article')
+];
+
+var inTree = [];
+var walker = collectNodes();
+var next = walker.iterate().next();
+while (!next.done) {
+	inTree.push(next.value);
+	next = walker.iterate().next();
+}
```

Comment:
This worked as a spreadable array until I realized we can't use Symbol or rest/spread.